yarl/__init__.pyi

```diff
@@ -33,7 +40,7 @@ class URL:
     @classmethod
     def build(cls, *, scheme: str=..., user: str=..., password: str=...,
               host: str=..., port: Optional[int]=..., path: str=...,
-              query: Optional[Mapping[str, str]]=..., query_string: str=...,
+              query: Query=..., query_string: str=...,
```

Comment:
I'd like to forbid `query="val"` in `build`.
Now it's technically possible but I consider it as a bug.
`query_string` exists and it is more obvious API.

Reply:
OK, if you want to prepare annotation for the future update, I can rewrite current solution as:

```
Query = Union[
    None,
    Mapping[str, QueryVariable],
    Sequence[Tuple[str, QueryVariable]]]

def build(cls, *, ...,
              query: Query=..., query_string: str=...)

    @overload
    def with_query(self, query: Union[str, Query]) -> URL: ...

    @overload
    def update_query(self, query: Union[str, Query]) -> URL: ...
```
Let me know if you prefere this solution.

But for me it looks confusing to have two query parameters for different types.
Maybe it is better to use only `query` parameter (of course with deprecation period) and incapsulate all the logic inside `build` function, like so:
```
    @classmethod
    def build(cls, *, scheme='', user='', password='', host='', port=None,
              path='', query=None, fragment='',
              encoded=False):

        if isinstance(query, str) and encoded:
            query_string = query
        else:
            query_string = cls._get_str_query(query)
       ....
```

Reply:
Deprecation `query_string` is a good idea maybe.
Would you make a PR for it?

Reply:
Sure